File: 1894-merge-strings-alternately/solution.c
#include<stdlib.h>
char * mergeAlternately(char * word1, char * word2)
{
    int n=strlen(word1);
    int m=strlen(word2);
    char *arr=(char *)malloc((n+m+1)*sizeof(char));
    int i=0,j=0,ind=0;
    while(i<n||j<m)
    {
        if(i<n)
        {
            arr[ind++]=word1[i++];
        }
        if(j<m)
        {
            arr[ind++]=word2[j++];
        }
    }
    arr[ind]='\0';
    return arr;
}

File: 1894-merge-strings-alternately/solution.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int n=word1.length();
        int m=word2.length();
        int i=0,j=0,ind=0;
        StringBuilder s=new StringBuilder();
        while(i<n || j<m)
        {
            if(i<n)
            {
                s.append(word1.charAt(i++));
            }
            if(j<m)
            {
                s.append(word2.charAt(j++));
            }
        }   
        return s.toString();
    }
}
